src/main/java/uk/gov/ch/oauth/tokens/SessionSignInModifier.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.ch.oauth.tokens;
+
+import java.util.Map;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.model.AccessToken;
+import uk.gov.companieshouse.session.model.UserProfile;
+
+public class SessionSignInModifier {
+
+    @SuppressWarnings("unchecked")
+    // This is necessary as the original data on the session is untyped, but expected to be of the correct types
+    static Map<String, Object> updateSignIn(final Object sInf, final Object sio) {
+        final Map<String, Object> original = (Map<String, Object>) sInf;
+        final Map<String, Object> extras = (Map<String, Object>) sio;
+        original.putAll(extras);
+        return original;
+    }
+
+    public void alterSessionData(Session session, AccessToken accessToken,
+            UserProfile userProfileResponse) {
+        SignInInfoMap signInInfo = new SignInInfoMap();
+        signInInfo.setUserProfile(userProfileResponse);
+        signInInfo.setAccessToken(accessToken);
+        signInInfo.setSignedIn(true);
+
+        alterSessionData(session.getData(), signInInfo.toMap());
+    }
+
+    public void alterSessionData(Map<String, Object> currentSessionData,
+            Map<String, Object> desiredSignInInfo) {
+
```

Comment:
Can get session data directly in this method instead of passing in `session.getData()`

Reply:
You're complaining about me passing in Session into the method that calls this. Could you please layout what you think the signatures should look like?

Reply:
Please see my comment above, we're getting the data from request attributes, I don't see the benefit of passing in the session when we can remove it from the method signature altogether and get the session within the method. Much like how it was done prior to your change.

```suggestion
public void alterSessionData(Map<String, Object> desiredSignInInfo) {
final Map<String, Object> sData = sessionFactory.getSessionDataFromContext();
```